src/mono/wasm/runtime/http.ts

```diff
@@ -41,14 +57,64 @@ export function http_wasm_abort_response(res: ResponseExtension): void {
     }
 }
 
+export function http_wasm_readable_stream_controller_enqueue(controller: ReadableByteStreamControllerExtension, bufferPtr: VoidPtr, bufferLength: number): void {
+    if (bufferLength === 0) {
+        controller.close();
+    } else {
+        // the bufferPtr is pinned by the caller
+        const view = new Span(bufferPtr, bufferLength, MemoryViewType.Byte);
+        const copy = view.slice() as Uint8Array;
+        if (controller.byobRequest) {
+            const v = controller.byobRequest.view as Uint8Array;
+            v.set(copy);
+            controller.byobRequest.respond(bufferLength);
+        } else {
+            controller.enqueue(copy);
+        }
+    }
+    controller.__resolve_pull();
+}
+
+export function http_wasm_readable_stream_controller_error(controller: ReadableByteStreamControllerExtension, error: Error): void {
+    controller.__reject_fetch(error);
+    controller.error(error);
+    controller.__resolve_pull();
+}
+
+export function http_wasm_fetch_stream(url: string, header_names: string[], header_values: string[], option_names: string[], option_values: any[], abort_controller: AbortController, pull: (controller: ReadableByteStreamControllerExtension, desired_size: number, pull_state: any) => void, pull_state: any): Promise<ResponseExtension> {
+    return new Promise((resolve_fetch, reject_fetch) => {
+        const body = new ReadableStream({
+            type: "bytes",
+            autoAllocateChunkSize: 65536,
+            pull(controller: ReadableByteStreamController) {
+                const { promise: pull_promise, promise_control: pull_promise_control } = createPromiseController<void>();
+                const c = controller as ReadableByteStreamControllerExtension;
+                let desired_size = c.desiredSize;
+                if (c.byobRequest && c.byobRequest.view) { // waiting on https://github.com/WebAssembly/design/issues/1162 so we can pass the view to .net
+                    desired_size = c.byobRequest.view.byteLength;
+                }
+                c.__resolve_pull = pull_promise_control.resolve;
+                c.__reject_fetch = reject_fetch;
+                pull(c, desired_size || 0, pull_state);
```

Comment:
is this `pull` calling itself ?

Reply:
I see that a parameter, could you please rename it so it doesn't clash ?

Reply:
No, it's calling the delegate which is passed into `http_wasm_fetch_stream`.

`pull: (controller: ReadableByteStreamControllerExtension, desired_size: number, pull_state: any) => void`